apps/cowswap-frontend/src/modules/injectedWidget/utils/validatePartnerFee.ts

```diff
@@ -19,9 +19,17 @@ export function validatePartnerFee(input: PartnerFee | undefined): string[] | un
   return errors.length > 0 ? errors : undefined
 }
 
-function validateRecipient(recipient: string): string | undefined {
+function validateRecipient(recipient: PartnerFee['recipient']): string | undefined {
   if (!recipient) return 'Partner fee recipient must be set!'
 
+  if (typeof recipient === 'string') return validateRecipientAddress(recipient)
+
+  const errors = Object.values(recipient).map(validateRecipientAddress).filter(isTruthy)
+
+  return errors.length > 0 ? errors.join(', ') : undefined
+}
+
+function validateRecipientAddress(recipient: string): string | undefined {
```

Comment:
Can you add a unit test for this fn?

Reply:
Sure, added: https://github.com/cowprotocol/cowswap/pull/4322/commits/0be10470707d318af1044e7ff83e5663e7b7fa0f